core/src/main/java/jenkins/agents/CloudSet.java

```diff
@@ -226,7 +226,9 @@ private void handleNewCloudPage(Descriptor<Cloud> descriptor, String name, Stapl
         checkName(name);
         JSONObject formData = req.getSubmittedForm();
         formData.put("name", name);
-        formData.put("cloudName", name); // ec2 uses that field name
+        // multiple cloud implementations use cloudName instead of name. It is simpler to retain this field rather than
+        // refactor all the downstream plugins and the related tests.
+        formData.put("cloudName", name);
```

Comment:
Simpler in the short term, yes, but what are the tradeoffs in terms of long-term maintenance?

Reply:
In the long term, I believe the maintenance extends to ensuring `Cloud.doConfigSubmit`, `Cloudset.handleNewCloudPage` and `_new.jelly` do not drop `cloudName`. I should also write a test to make sure a cloud with `cloudName` successfully creates a new cloud to ensure there are now regressions.

I was concerned (maybe overly) about changing a public constructor variable `cloudName` to `name` in `ec2` and `azure-vm-agents`, which then causes breakages in existing casc files.

Reply:
Is there a way to support both `cloudName` and `name` in `ec2` and `azure-vm-agents` for a transition period, enabling users to upgrade and migrate to the new usage before eventually dropping support for `cloudName`?

Reply:
I wish I could deprecate the constructor, but as I'm just changing the variable name, it's not really helping.
I don't know if there is something I can add in CasC as a special case either.

Reply:
for configuration as code the easiest way is to make a breaking change if you want to change a required variable name, it's annoying but it fails-fast for the users with clear instructions in release notes it's not too bad.

you could hack around it with a databoundsetter for the cloud name and have jcasc usage support that and then check both values to calculate the right one but I don't think it's worth it.

Reply:
Given that a smooth migration path is not practical, my vote would be to bite the bullet and make this breaking change now rather than to have to carry this workaround forever.

Reply:
Changes filed here:
https://github.com/jenkinsci/azure-vm-agents-plugin/pull/462
https://github.com/jenkinsci/ec2-plugin/pull/892

Reply:
removed `cloudName` references in 231db9490cbc18d035805eaee4f8010fdfa0472b